src/tests/Interop/PInvoke/Variant/VariantTest.ComWrappers.cs

```diff
@@ -0,0 +1,112 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using TestLibrary;
+using static VariantNative;
+using ComTypes = System.Runtime.InteropServices.ComTypes;
+
+#pragma warning disable CS0612, CS0618
+partial class Test
+{
+    public static int Main()
+    {
+        bool testComMarshal=true;
+        ComWrappers.RegisterForMarshalling(new ComWrappersImpl());
+        try
+        {
+            TestByValue(testComMarshal);
+            TestByRef(testComMarshal);
+            TestOut();
+            TestFieldByValue(testComMarshal);
+            TestFieldByRef(testComMarshal);
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine($"Test failed: {e}");
+            return 101;
+        }
+        return 100;
+    }
+}
+
+internal unsafe class ComWrappersImpl : ComWrappers
+{
+    private static readonly ComInterfaceEntry* wrapperEntry;
+
+    static ComWrappersImpl()
+    {
+        var vtblRaw = (IntPtr*)RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IDispatchVtbl), sizeof(IntPtr) * 7);
+        GetIUnknownImpl(out vtblRaw[0], out vtblRaw[1], out vtblRaw[2]);
+
+        vtblRaw[3] = (IntPtr)(delegate* unmanaged<IntPtr, IntPtr, int>)&IDispatchVtbl.GetTypeInfoCountInternal;
+        vtblRaw[4] = (IntPtr)(delegate* unmanaged<IntPtr, int, int, IntPtr, int>)&IDispatchVtbl.GetTypeInfoInternal;
+        vtblRaw[5] = (IntPtr)(delegate* unmanaged<IntPtr, IntPtr, IntPtr, int, int, IntPtr, int>)&IDispatchVtbl.GetIDsOfNamesInternal;
+        vtblRaw[6] = (IntPtr)(delegate* unmanaged<IntPtr, int, IntPtr, int, ComTypes.INVOKEKIND, IntPtr, IntPtr, IntPtr, IntPtr, int>)&IDispatchVtbl.InvokeInternal;
+
+        wrapperEntry = (ComInterfaceEntry*)RuntimeHelpers.AllocateTypeAssociatedMemory(typeof(IDispatchVtbl), sizeof(ComInterfaceEntry));
+        wrapperEntry->IID = new Guid("00020400-0000-0000-C000-000000000046");
```

Comment:
Let's make this a local and name it appropriately: `IID_IDispatch`.

Reply:
Actually putting this as a static on the `IDispatchVtbl` would be my preference.